pkg/certificate/authorization/service.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package authorization
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"storj.io/storj/pkg/pb"
+)
+
+// ErrService is the default error class for the authorization service.
+var ErrService = errs.Class("authorization service error")
+
+// Service is the authorization service.
+type Service struct {
+	log *zap.Logger
+	db  *DB
+}
+
+// NewService creates a new authorization service.
+func NewService(log *zap.Logger, db *DB) *Service {
+	return &Service{
+		log: log,
+		db:  db,
+	}
+}
+
+// GetOrCreate will return an authorization for the given user ID
+func (service *Service) GetOrCreate(ctx context.Context, req *pb.AuthorizationRequest) (_ *pb.AuthorizationResponse, err error) {
```

Comment:
Oh, that's what I commented.
For me, the services shouldn't depend on protocol buffers, but I've seen several services that they do, so I thought that it is the way to go in Storj.

Reply:
It's more of we haven't fixed everything, but preference should be to avoid them, since it ties our internal code to protobuf, which makes fixes to protobuf harder.

Reply:
> It's more of we haven't fixed everything,

Good to know, so I can point to that in new services.

> but preference should be to avoid them, since it ties our internal code to protobuf, which makes fixes to protobuf harder.

Exactly that, it's because I said that "in my opinion, they shouldn't depend on protobuf".
Thanks for the clarification.

I'm going to improve https://github.com/storj/docs/blob/master/code/Service.md

Reply:
Just to clarify there are some messages that need to use protobuf for serialization, like `pb.OrderLimit` since the signature is based on it... and we don't have a good alternative at the moment. But for new services we should prefer either new structs or regular arguments.

Reply:
That's the example in the document, however we should think in replace the example by one that it doesn't make any exception and make the people raise the point when they encounter one.

Thanks for the clarification.

Reply:
https://github.com/storj/docs/pull/110